cognite/client/data_classes/transformations/__init__.py

```diff
@@ -202,34 +208,31 @@ def try_get_or_create_nonce(oidc_credentials: OidcCredentials | None, project: s
             # MyPy requires this to make sure it's not changed to None after inner declaration
             assert sessions_cache is not None
 
-            key = (
-                f"{oidc_credentials.client_id}:{hash(oidc_credentials.client_secret)}:{project}"
-                if oidc_credentials
-                else "DEFAULT"
-            )
+            key = "DEFAULT"
+            if oidc_credentials:
+                key = f"{oidc_credentials.client_id}:{hash(oidc_credentials.client_secret)}:{project}"
 
-            ret = sessions_cache.get(key)
-            if not ret:
+            if (ret := sessions_cache.get(key)) is None:
+                credentials = None
                 if oidc_credentials and oidc_credentials.client_id and oidc_credentials.client_secret:
                     credentials = ClientCredentials(oidc_credentials.client_id, oidc_credentials.client_secret)
-                else:
-                    credentials = None
-                try:
-                    session = self._cognite_client.iam.sessions.create(credentials, project=project)
-                    ret = NonceCredentials(session.id, session.nonce, project)
-                    sessions_cache[key] = ret
-                except Exception:
-                    ret = None
+
+                with patch_attribute(self._cognite_client.config, "project", project):
+                    try:
+                        session = self._cognite_client.iam.sessions.create(credentials)
+                        ret = sessions_cache[key] = NonceCredentials(session.id, session.nonce, project)
+                    except CogniteAPIError as err:
+                        logger.debug(f"Unable to create a session and get a nonce towards {project=}: {err!r}")
```

Comment:
@erlendvollset , is this useful or not? (Seems wrong to hide inside a catch-all at least 🤔 )